hathor/transaction/storage/transaction_storage.py

```diff
@@ -575,7 +600,7 @@ def _save_transaction(self, tx: BaseTransaction, *, only_metadata: bool = False)
     def _reset_cache(self) -> None:
         """Reset all caches. This function should not be called unless you know what you are doing."""
         if not self.with_index:
-            raise NotImplementedError
+            raise RuntimeError('Cannot reset cache because it has not been enabled.')
```

Comment:
Why not just `assert self.with_index, 'Cannot reset cache because it has not been enabled.'`? Is this a situation we can actually reach except from a miss-configured code setup?

Reply:
Exactly. That's why I chose RuntimeError, but we can use an assert statement as well.

Reply:
Changed! Can you check again, please?

Reply:
Looks good.